src/lib/utils/hooks/SynapseAPI/forum/useForum.ts

```diff
@@ -54,18 +61,128 @@ export function useGetForumInfinite(
   )
 }
 
-export function useGetThread(
+export function useGetThread(threadId: string) {
+  const { data: threadData, isLoading: isLoadingBundle } =
+    useGetThreadBundle(threadId)
+  const { data: threadBody, isLoading: isLoadingBody } = useGetThreadBody(
+    threadData?.messageKey ?? '',
+  )
+  const isLoading = isLoadingBody || isLoadingBundle
+  return { threadData, threadBody, isLoading }
+}
+
+export function useGetThreadBundle(
   threadId: string,
   options?: UseQueryOptions<DiscussionThreadBundle, SynapseClientError>,
 ) {
   const { accessToken } = useSynapseContext()
   return useQuery<DiscussionThreadBundle, SynapseClientError>(
-    ['thread', threadId],
+    ['thread', threadId, accessToken],
     () => SynapseClient.getThread(threadId, accessToken),
     options,
   )
 }
 
+export function useGetThreadBody(
+  messageKey: string,
+  options?: UseQueryOptions<string, SynapseClientError>,
+) {
+  const { accessToken } = useSynapseContext()
+  const messageUrl = SynapseClient.getThreadMessageUrl(messageKey, accessToken)
+  return useQuery<string, SynapseClientError>(
+    ['thread', messageKey, accessToken],
+    async () => getMessage((await messageUrl).messageUrl),
+    options,
+  )
+}
```

Comment:
Consistently use the thread ID in the second field.

You could also define a type and use an object instead of an array to keep this organized. If you want to do that, take a look at queryKeys.ts which is used for entities.

```suggestion
export function useGetThreadBody(
  threadBundle: DiscussionThreadBundle,
  options?: UseQueryOptions<string, SynapseClientError>,
) {
  const { accessToken } = useSynapseContext()
  const messageUrl = SynapseClient.getThreadMessageUrl(messageKey, accessToken)
  return useQuery<string, SynapseClientError>(
    ['thread', threadBundle.id, messageKey, accessToken],
    async () => getMessage((await messageUrl).messageUrl),
    options,
  )
}
```